File: software/examples/perl/example_simple.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletColor;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'abc'; # Change to your UID

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $c = Tinkerforge::BrickletColor->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get current color
my ($r, $g, $b, $cl) = $c->get_color();

print "Color(R): $r\n";
print "Color(G): $g\n";
print "Color(B): $b\n";
print "Color(C): $cl\n";
print "\n";

print "Press any key to exit...\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_simple.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_color'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Color Bricklet

ipcon = IPConnection.new # Create IP connection
c = BrickletColor.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get current color as [r, g, b, c]
color = c.get_color

puts "Color [R]: #{color[0]}"
puts "Color [G]: #{color[1]}"
puts "Color [B]: #{color[2]}"
puts "Color [C]: #{color[3]}"

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
